src/Explorer/Panes/LoadQueryPanel/index.tsx

```diff
@@ -0,0 +1,147 @@
+import Explorer from "../../Explorer";
+import { ConsoleDataType } from "../../Menus/NotificationConsole/NotificationConsoleComponent";
+import QueryTab from "../../Tabs/QueryTab";
+import { Collection } from "..//../../Contracts/ViewModels";
+import { GenericRightPaneComponent, GenericRightPaneProps } from "../GenericRightPaneComponent";
+
+interface LoadQueryPanelProps {
+  explorer: Explorer;
+  closePanel: () => void;
+}
+
+export const LoadQueryPanel: FunctionComponent<LoadQueryPanelProps> = ({
+  explorer,
+  closePanel,
+}: LoadQueryPanelProps): JSX.Element => {
+  const [isLoading, { setTrue: setLoadingTrue, setFalse: setLoadingFalse }] = useBoolean(false);
+  const [formError, setFormError] = useState<string>("");
+  const [formErrorsDetails, setFormErrorsDetails] = useState<string>("");
+  const [selectedFileName, setSelectedFileName] = useState<string>("");
+  const [selectedFiles, setSelectedFiles] = useState<FileList>();
+
+  const imageProps: Partial<IImageProps> = {
+    imageFit: ImageFit.centerCover,
+    width: 20,
+    height: 20,
+    className: "fileIcon",
+  };
+
+  const title = "Load Query";
+  const genericPaneProps: GenericRightPaneProps = {
+    container: explorer,
+    formError: formError,
+    formErrorDetail: formErrorsDetails,
+    id: "loadQueryPane",
+    isExecuting: isLoading,
+    title,
+    submitButtonText: "Load",
+    onClose: () => closePanel(),
+    onSubmit: () => submit(),
+  };
+
+  const onFileSelected = (e: React.ChangeEvent<HTMLInputElement>): void => {
+    const { files } = e.target;
+    setSelectedFiles(files);
+    setSelectedFileName(files && files[0] && `"${files[0].name}"`);
+  };
+
+  const submit = (): void => {
+    setFormError("");
+    setFormErrorsDetails("");
+    if (!selectedFiles || selectedFiles.length === 0) {
+      setFormError("No file specified");
+      setFormErrorsDetails("No file specified. Please input a file.");
+      logConsoleError("Could not load query -- No file specified. Please input a file.");
+      return;
+    }
+
+    const file: File = selectedFiles[0];
+    const id: string = logConsoleMessage(ConsoleDataType.InProgress, `Loading query from file ${file.name}`);
+    logConsoleProgress(`Loading query from file ${file.name}`);
+    setLoadingTrue();
+    loadQueryFromFile(file)
+      .then(
+        () => {
+          logConsoleInfo(`Successfully loaded query from file ${file.name}`);
+          closePanel();
+        },
+        (error: Error) => {
+          setFormError("Failed to load query");
+          setFormErrorsDetails(`Failed to load query: ${error}`);
+          logConsoleError(`Failed to load query from file ${file.name}: ${error}`);
+        }
+      )
+      .finally(() => {
+        setLoadingFalse();
+        clearInProgressMessageWithId(id);
+      });
+  };
+
+  const loadQueryFromFile = async (file: File): Promise<void> => {
+    const selectedCollection: Collection = explorer && explorer.findSelectedCollection();
```

Comment:
nit: use optional chaining where possible:
```suggestion
    const selectedCollection: Collection = explorer?.findSelectedCollection();
```

Reply:
updated